app/src/main/java/com/example/triptracker/view/profile/FriendListView.kt

```diff
@@ -0,0 +1,158 @@
+                    style = AppTypography.secondaryTitleStyle,
+                )
+                Spacer(modifier = Modifier.width(2.dp))
+                Text(
+                    text = friend.surname,
+                    style = AppTypography.secondaryTitleStyle,
+                )
+              }
+        }
+        Column(
+            modifier = Modifier.fillMaxWidth().testTag("Friend"),
+            horizontalAlignment = Alignment.End) {
+              if (followers) {
+                // Display the remove follower button
+                RemoveFriendButton(
+                    remove = { userProfileViewModel.removeFollowerInDb(userProfile, friend) },
+                    undoRemove = { userProfileViewModel.addFollowersInDb(userProfile, friend) },
+                    followers = true)
+              } else {
+                RemoveFriendButton(
+                    remove = { userProfileViewModel.removeFollowingInDb(userProfile, friend) },
+                    undoRemove = { userProfileViewModel.addFollowingInDb(userProfile, friend) },
+                    followers = false)
+              }
+            }
+      }
+    }
+  }
+}
+
+/** This composable function displays a button to remove a follower. */
+@Composable
+fun RemoveFriendButton(remove: () -> Unit, undoRemove: () -> Unit, followers: Boolean) {
+  // State to determine if the follower is removed and update the button text
+  var isRemoved by remember { mutableStateOf(false) }
+
+  Button(
+      onClick = {
+        isRemoved = !isRemoved
+        // Remove the follower if the button is clicked
+        if (isRemoved) {
+          remove()
+        }
+        // Undo the removal if the button is clicked again
+        else {
+          undoRemove()
+        }
+      },
+      colors =
+          if (!isRemoved) {
+            ButtonDefaults.buttonColors(
+                containerColor = md_theme_light_dark,
+                contentColor = md_theme_light_secondaryContainer)
+          } else {
+            ButtonDefaults.buttonColors(
+                containerColor = md_theme_light_secondaryContainer,
+                contentColor = md_theme_light_dark)
+          },
+      modifier =
+          Modifier.height(40.dp)
+              .width(120.dp)
+              .testTag("RemoveButton")) { // .testTag(if (isRemoved) { "RemoveButton" } else {
+        // "UndoButton" })) {
+        Text(
+            text =
+                // Display the appropriate button text based on whether we are prompting
+                // follower or following and whether the friend have been removed or not
+                if (!followers && isRemoved) "Follow" // when displaying removed following
+                else if (!followers && !isRemoved) "Following" // when displaying following
+                else if (followers && isRemoved) "Undo" // when displaying removed followers
+                else "Remove" // when displaying followers
+            )
+      }
+}
```

Comment:
Is there a preview for this screen ? So we can look at it individually ?

Reply:
I didn't add a preview but I will then